src/components/Modal/ModalHeader.test.tsx

```diff
@@ -124,16 +157,27 @@ describe('ModalHeader', () => {
   })
 
   describe('close button', () => {
-    it('should render close button when "hasCloseIcon" is true', () => {
+    it('should render close button when "hasCloseButton" is true and modal is called with props', () => {
       render(
         <ModalContextProvider value={mockContextValue}>
-          <ModalHeader title='title' hasCloseIcon />
+          <ModalHeader title='title' hasCloseButton />
         </ModalContextProvider>
       )
       expect(screen.getByRole('button')).toBeInTheDocument()
     })
 
-    it('should render close button when "hasCloseIcon" is not provided (default true)', () => {
+    it('should render close button when "hasCloseButton" is true and modal is called with children', () => {
```

Comment:
What is the difference between this test and the test that starts in line 189?

"_should render close button when "hasCloseButton" is not provided (default true) and modal is called with children_"

Reply:
The idea was to keep the two previous tests comparing the behaviors between passing true and not passing (`true` is default), but the implementation of the first test was actually wrong. I passed the prop as true to make sense.